sale_exception_product_sale_manufactured_for/models/sale_order.py

```diff
@@ -0,0 +1,46 @@
+# Copyright 2021 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+from odoo import models
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    def exception_partner_can_order_manufactured_for(self):
+        """Return whether the partner can order all the items in the sale
+
+        Used in a sales exception. Returns true only if the partner set in
+        the sales order can buy all the products. This will happen if:
+        1. it has a commercial entity that is listed as one of the
+           partners for which the products can be manufactured for.
+        2. it doesn't have a commercial entity and none of the products
+           on the sale order have any restriction.
+        """
+        self.ensure_one()
+        commercial_entity = self.partner_id.commercial_partner_id
+
+        self.env.cr.execute(
+            """
+            SELECT rel.product_id, ARRAY_AGG(rel.partner_id)
```

Comment:
Is it possible to use `SQL` class to wrap query?

Reply:
will fix